addons/mass_mailing/models/mail_mail.py

```diff
@@ -86,29 +78,32 @@ def _sub_relative2absolute(match):
         body = re.sub(r'(<[^>]+\bstyle="[^"]+\burl\(\'?)(/[^/\'][^\'")]+)', _sub_relative2absolute, body)
 
         # generate tracking URL
-        if mail.statistics_ids:
-            tracking_url = self._get_tracking_url(cr, uid, mail, partner, context=context)
+        if self.statistics_ids:
+            tracking_url = self._get_tracking_url(partner)
             if tracking_url:
                 body = tools.append_content_to_html(body, tracking_url, plaintext=False, container_tag='div')
         return body
 
-    def send_get_email_dict(self, cr, uid, ids, partner=None, context=None):
+    @api.multi
+    def send_get_email_dict(self, partner=None):
         # TDE: temporary addition (mail was parameter) due to semi-new-API
-        res = super(MailMail, self).send_get_email_dict(cr, uid, ids, partner, context=context)
-        mail = self.browse(cr, uid, ids[0], context=context)
-        base_url = self.pool.get('ir.config_parameter').get_param(cr, uid, 'web.base.url')
-        if mail.mailing_id and res.get('body') and res.get('email_to'):
+        self.ensure_one()
```

Comment:
@tivisse ,
in old code it was always take single record id.. 
see 
mail = self.browse(cr, uid, ids[0], context=context)

that's why we specified ensure_one()